### PR TITLE
initial_commit_hw_#3

### PR DESCRIPTION
**Pull request by harryji168**
Homework 3 with stretch

test data: 
> s
Where? (myTodos.json)
>tmp/a/apple
List saved to "tmp/a/apple/myTodos.json"

### CODE REVIEW

File: Todo_CLI/todoCLIWithStretch.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+function ensureDirectoryExistence(filePath) {
+    var dirname = path.dirname(filePath);
+    if (fs.existsSync(dirname)) {
+      return true;
+    }
+    ensureDirectoryExistence(dirname);
+    fs.mkdirSync(dirname);
+}
+function write_fie(){
```
Comment (kaylyn94) [inline Todo_CLI/todoCLIWithStretch.js:30]: typo

File: Todo_CLI/todoCLIWithStretch.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        if (err) {
+            console.log('Could not write to file');
+            console.error(err);
+            return;
+        }
+        console.log(`List saved to "${fileName}"`);
+        MainMenu();
+    });    
+}
+function MainMenu() {
```
Comment (kaylyn94) [inline Todo_CLI/todoCLIWithStretch.js:41]: A function is an action so should start with a verb and don't start it with a capital letter. It's the convention for class, not function.

File: Todo_CLI/todoCLIWithStretch.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+///////////////////////////////////////////////////////////////////////////////////
```
Comment (kaylyn94) [inline Todo_CLI/todoCLIWithStretch.js:1]: Do not produce duplicate code just for an additional feature. Stretch should be done in one file.

File: Todo_CLI/todoCLIWithStretch.js (hunk 1 above)
Comment (kaylyn94) [inline Todo_CLI/todoCLIWithStretch.js:30]: expecting `fileName` as an argument

**Comment by kaylyn94** [comment]
# TODO CLI HOMEWORK 3
## SCORESHEET FOR STUDENT Harry

### Write an interactive CLI todo list application using Node's readline and fs modules. 

### GRADING CRITERIA:
### Course Value: 5%
### Assignment Value: 100%

### GRADE:
|Total Grade:||85|
|----------|----|----|
|Score| 85 | /100|
|Deductions| 00 |/20|

***LATE ASSIGNMENTS WILL BE DEDUCTED 5% OFF PER 24 HOURS, FOR A MAX OF 20% DEDUCTION OVERALL.
YOUR ASSIGNMENT IS CONSIDERED LATE 5 MINUTES AFTER THE DUE DATE AND TIME, UNLESS OTHERWISE DISCUSSED WITH YOUR INSTRUCTOR.***
__________________________________________________________
__________________________________________________________

### Part 1: Submitted and on Time

> Was the assignment complete and on time?

|Feature |Mark |Notes |
|--------|-----|------|
|Complete and submitted on time.|✅|All good!|
|On time, but incomplete.|||
|Complete, but submitted late.|||
|Incomplete and submitted late.|||
|Very late submission|||
|No submission|||

### Part 2: Features

> Was everything in assignment attempted?

|Feature |Mark |Notes |
|--------|-----|------|
|Implemented the Menu |✅||
|Implemented View |✅||
|Implemented Add/New |✅||
|Implemented Complete cX and mark completed item |✅||
|User notified which item was marked with cX |✅||
|Implemented Delete |✅||
|User notified which item was deleted |✅||
|Implemented Quit |✅||

### Part 3: Stretches

> Were Stretches attempted?

|Feature |Mark |Notes |
|--------|-----|------|
|Attempted Stretch 1.|❌||
|Attempted Stretch 2.|❌||

__________________________________________________________
__________________________________________________________

### ASSIGNMENT RUBRIC

|A+|A|A-|B+|B|B-|C+|C-F|
|--------|-------|-------|-------|-------|-------|-------|------|
|90%-100%|85%-89%|80%-84%|76%-79%|72%-75%|68%-71%|64%-67%|0%-63%|
|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  All Stretches or Challenges were completed.  Styling and design of front end meets requirements, and extra creative effort has been applied to this assignment overall.|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable. **OR** Stretches or Challenges not attempted, but creative effort has been applied to the assignment overall.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created and works correctly or mostly works.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time, but handed in as incomplete.  Most instructions were followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Some required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Minimal or no effort.Incomplete/ missing components/insufficient.|

File: Todo_CLI/todoCLIWithStretch.js (hunk 2 above)
Comment (harryji168) [reply Todo_CLI/todoCLIWithStretch.js:41]: thanks

File: Todo_CLI/todoCLIWithStretch.js (hunk 3 above)
Comment (harryji168) [reply Todo_CLI/todoCLIWithStretch.js:1]: ok, thanks

File: Todo_CLI/todoCLIWithStretch.js (hunk 1 above)
Comment (harryji168) [reply Todo_CLI/todoCLIWithStretch.js:30]: yes

Comment (harryji168) [reply Todo_CLI/todoCLIWithStretch.js:30]: what "typo" mean?

**Pull request merged**